mlrun/artifacts/manager.py

```diff
@@ -315,7 +315,13 @@ def log_artifact(
             item.upload(artifact_path=artifact_path)
 
         if db_key:
-            self._log_to_db(db_key, project, producer.inputs, item)
+            artifact_item = self._log_to_db(db_key, project, producer.inputs, item)
+            if artifact_item:
+                artifact_uid = artifact_item.get("metadata", {}).get("uid")
+                if artifact_uid is not None:
+                    item.uid = artifact_uid
+                    self.artifact_uris[key] = item.uri
```

Comment:
Why not return the `uid` from `_log_to_db` instead of the full item, since it's only used here.